bookkeeper-server/src/main/java/org/apache/bookkeeper/client/PendingAddOp.java

```diff
@@ -91,14 +88,11 @@ static PendingAddOp create(LedgerHandle lh, ByteBuf payload, AddCallback cb, Obj
         op.completed = false;
         op.ackSet = lh.distributionSchedule.getAckSet();
         op.addOpLogger = lh.bk.getAddOpLogger();
-        if (op.timeout != null) {
-            op.timeout.cancel();
-        }
-        op.timeout = null;
-        op.timeoutSec = lh.bk.getConf().getAddEntryQuorumTimeout();
+        op.timeoutNanos = TimeUnit.SECONDS.toNanos(lh.bk.getConf().getAddEntryQuorumTimeout());
         op.pendingWriteRequests = 0;
         op.callbackTriggered = false;
         op.hasRun = false;
+        op.requestTimeNanos = Long.MAX_VALUE;
```

Comment:
did I miss something there? why the requestTimeNanos is `Long.MAX_VALUE`?

Reply:
It's defensively defaulting to disabling the timeout if we forget to set the start time of the request. Long.MAX_VALUE will always be in the future.